### PR TITLE
Feature/ccclxxxvi 6

### PR DESCRIPTION
**Pull request by ghost**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=993)

## Code reviewers

- [ ] @Andriy97091
- [x] @irynazaiets
- [x] @jelysaw
- [x] @MaksymGromyk
- [ ] @shado1997
- [ ] @tavor118
- [x] @VolodymyrBodnar
- [x] @wm139

### Second Level Review

- [ ] @lhalam
- [ ] @alutsysh

## Summary of issue
create pool manager for our DB
ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by vengvas — DISMISSED** [review]

**Review by jelysaw — DISMISSED** [review]

File: moneta/src/python/core/db/pool_manager.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+@singleton
+class DBPoolManager:
+    '''Class for managing connections to the DB'''
+    def __init__(self):
+        '''creating new instance of DB _pool manager'''
+        data = utils.get_db_config()
+        self._connection_counter = 0
+        self._pool = []
+        self.lock = threading.RLock()
```
Comment (alutsysh) [inline moneta/src/python/core/db/pool_manager.py]: do you need all these values as public?

File: moneta/src/python/core/db/pool_manager.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                                     user=self.user,
+                                     password=self.password,
+                                     port=self.port)
+        self._connection_counter += 1
+        return {CONNECTION: connection,
+                LAST_UPDATE: 0,
+                CREATE_TIME: time.time()}
+
+    def _get_connection(self):
+        '''get connection from the connection _pool'''
```
Comment (alutsysh) [inline moneta/src/python/core/db/pool_manager.py]: capitalize the first letter

File: moneta/src/python/core/db/pool_manager.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        connection[CONNECTION].close()
+        self._connection_counter -= 1
+
+    def _return_connection(self, connection):
+        '''return connection to the _pool'''
+        connection[LAST_UPDATE] = time.time()
+        self._pool.append(connection)
+
+    @contextmanager
+    def manage(self):
```
Comment (alutsysh) [inline moneta/src/python/core/db/pool_manager.py]: Rename?

File: moneta/src/python/core/db/pool_manager.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        self.__connection_counter = 0
+        self.__pool = []
+        self.__lock = threading.RLock()
+        self.__database = data['moneta']['database']
+        self.__user = data['moneta']['user']
+        self.__port = data['moneta']['port']
+        self.__password = data['moneta']['password']
+        self.__lifetime = data['moneta']['lifetime']
+        self.__delay = data['moneta']['delay']
+        self.___poolsize = data['moneta']['poolsize']
```
Comment (alutsysh) [inline moneta/src/python/core/db/pool_manager.py]: new type?) 3*_

Comment (ghost) [reply moneta/src/python/core/db/pool_manager.py]: super private - 'I will cut your hands'

**Review by alutsysh — APPROVED** [review]

**Review by irynazaiets — APPROVED** [review]

**Review by VolodymyrBodnar — APPROVED** [review]

**Review by MaksymGromyk — APPROVED** [review]

**Pull request merged**